### PR TITLE
waitready: Add remediation to error message

### LINKED ISSUE #6790: wait_for_integration error message is unclear
When apm-server runs standalone, it will (by default) wait for the integration package to be installed. We check this is two ways: by querying the Fleet API, or by checking that index templates have been installed. The error message is not super clear:

> {"log.level":"error","@timestamp":"2021-12-03T14:02:24.454-0500","log.logger":"beater","log.origin":{"file.name":"beater/waitready.go","file.line":58},"message":"error checking preconditions: error querying Elasticsearch for integration index templates: unexpected HTTP status: 404 Not Found ({\"error\":{\"root_cause\":[{\"type\":\"resource_not_found_exception\",\"reason\":\"index template matching [logs-apm.error] not found\"}],\"type\":\"resource_not_found_exception\",\"reason\":\"index template matching [logs-apm.error] not found\"},\"status\":404})","service.name":"apm-server","ecs.version":"1.6.0"}

We should:

  1. Clarify that the server is waiting for the integration package to be installed.
  2. Direct the user to install the integration package to resolve the issue.

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

Adds some more context and information to the `checkintegration` check.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. `docker-compose up -d`
2. `make`
3. `./apm-server -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -e` outputs a log message with `to remediate, please install the apm integration`.
4. `./apm-server -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.kibana.host=http://localhost:5601 -E apm-server.kibana.username=admin -E apm-server.kibana.password=changeme -E apm-server.kibana.enabled=true -e` outputs a log message with `to remediate, please install the apm integration`.

## Related issues

Closes #6790

### CODE REVIEW

**Review by axw — APPROVED** [review]

File: beater/checkintegration.go [hunk 1]
```diff
@@ -42,7 +42,17 @@ func checkIntegrationInstalled(
...
 	logger *logp.Logger,
-) error {
+) (err error) {
+	defer func() {
+		if err != nil {
+			// We'd like to include some remediation actions when the APM Integration isn't installed.
+			err = &actionableError{
+				Err:         err,
+				Name:        "apm integration installed",
+				Remediation: "please install the apm integration",
```
Comment (simitt) [inline beater/checkintegration.go:52]: Generally +1 on introducing `actionableError`, but not certain how helpful this exact Remediation note is to users that are not familiar to the term `apm integration` or don't know how to install it. Ideally, if we are confident that doc links don't change frequently, we could link to the docs from here.

Comment (axw) [reply beater/checkintegration.go:52]: We could add an ela.st shortened link (https://links.elastic.dev/), which can be edited. WDYT?

Comment (simitt) [reply beater/checkintegration.go:52]: That sounds perfect!

Comment (marclop) [reply beater/checkintegration.go:52]: Is this the correct URL? https://www.elastic.co/guide/en/apm/server/current/apm-integration.html

**Review by simitt — APPROVED** [review]

**Comment by stuartnelson3** [comment]
step 3 confirmed with https://github.com/elastic/apm-server/commit/2d1c27052b5ddce37858c85b48a84561752f791d 

step 4 resulted in:
```json
{"log.level":"info","@timestamp":"2021-12-22T15:48:07.567+0100","log.logger":"beater","log.origin":{"file.name":"beater/checkintegration.go","file.line":105},"message":"integration package status: installed","service.name":"apm-server","ecs.version":"1.6.0"}
```

Going to mark this as `test-plan-regression` for now @marclop

**Comment by marclop** [comment]
I was able to verify step `4` with 2d1c27052:

1. `docker-compose down`
2. `docker-compose up -d`
3. `make`
4. `./apm-server -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.kibana.host=http://localhost:5601 -E apm-server.kibana.username=admin -E apm-server.kibana.password=changeme -E apm-server.kibana.enabled=true -e`

```json
{"log.level":"error","@timestamp":"2021-12-23T10:48:55.109+0800","log.logger":"beater","log.origin":{"file.name":"beater/waitready.go","file.line":62},"message":"precondition 'apm integration installed' failed: integration package not yet installed: to remediate, please install the apm integration: https://ela.st/apm-integration-quickstart","service.name":"apm-server","ecs.version":"1.6.0"}
```

**Comment by stuartnelson3** [comment]
Verified with 2d1c27052 using `docker-compose up` in the `apm-server` repo.

The message I got above (`integration package status: installed`) occurs when starting up `apm-integration-testing`, but without explicitly installing the package. I'm not sure if this is expected or something we should be concerned with.

**Comment by marclop** [comment]
@stuartnelson3 I believe it's expected since this commit was merged: https://github.com/elastic/apm-integration-testing/commit/6c27fc8fc88e83ae6ebf223875b2e18600006a81

**Pull request merged**